bootstore/src/schemes/v0/storage.rs

```diff
@@ -96,10 +97,54 @@ pub struct NetworkConfig {
 
 impl Ledgerable for NetworkConfig {
     fn is_newer_than(&self, other: &Self) -> bool {
-        self.generation >= other.generation
+        self.generation > other.generation
     }
 
     fn generation_bump(&mut self) {
         self.generation += 1;
     }
 }
+
+impl NetworkConfig {
+    /// Save the `NetworkConfig` to a ledger.
+    ///
+    /// Panics if the ledger cannot be saved.
+    pub async fn save(
+        log: &Logger,
+        paths: Vec<Utf8PathBuf>,
+        mut config: NetworkConfig,
+    ) {
+        // Decrement the generation number of the config, so that when we commit
+        // it the commit bumps it back to the actual generation number. This
+        // keeps the generation number of the network config identical to the
+        // ledger generation.
+        //
+        // Note that generation numbers of valid configs must start at 1, so we
+        // specifically do not use a saturating subtraction.
+        config.generation -= 1;
```

Comment:
In that case, maybe `checked_sub(1).unwrap()` as an extra check?

Reply:
wait, does this not panic in release mode? Shit. Good call.

Reply:
Unless the release build it built with [overflow checks](https://rust-lang.github.io/rfcs/1535-stable-overflow-checks.html); we do this in Hubris, but I'm not sure about Omicron.

Reply:
Done in [cd37295](https://github.com/oxidecomputer/omicron/pull/3612/commits/cd3729569e81ebd495999a73ef5ee06a03af3085)